1.6/windows/nanoserver/Dockerfile

```diff
@@ -0,0 +1,53 @@
+FROM microsoft/nanoserver
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop';"]
+
+# ideally, this would be C:\go to match Linux a bit closer, but C:\go is the recommended install path for Go itself on Windows
+ENV GOPATH C:\\gopath
+
+# PATH isn't actually set in the Docker image, so we have to set it from within the container
+RUN Write-Host 'Updating PATH ...'; \
+	$regPath = 'Registry::HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment'; \
+	\
+	$oldPath = ( \
+	Get-ItemProperty \
+		-Path $regPath \
+		-Name PATH \
+	).path; \
+	Write-Host ('  Before: {0}' -f $oldPath); \
+	\
+	$newPath = $env:GOPATH + '\bin;C:\go\bin;' + $oldPath; \
+	Write-Host ('  After:  {0}' -f $newPath); \
+	\
+	Set-ItemProperty \
+		-Path $regPath \
+		-Name PATH \
+		-Value $newPath; \
+	Write-Host 'Complete.';
+# doing this first to share cache across versions more aggressively
+
+ENV GOLANG_VERSION 1.6.3
+ENV GOLANG_DOWNLOAD_URL https://golang.org/dl/go$GOLANG_VERSION.windows-amd64.zip
+ENV GOLANG_DOWNLOAD_SHA256 6a18e5ed8b39785338986aecc6a3f36f5c4be286ff52db0ae3bcd2275ab70df0
+
+RUN Write-Host ('Downloading {0} ...' -f $env:GOLANG_DOWNLOAD_URL); \
+	Invoke-WebRequest -Uri $env:GOLANG_DOWNLOAD_URL -OutFile 'go.zip'; \
```

Comment:
I ran the following `Measure-Command` lines several times, and they came up pretty comparable on average:

```powershell
$url = 'http://ovh.net/files/100Mb.dat'

# ~4s
Measure-Command { Invoke-WebRequest -Uri $url | Out-Null }

# ~4s
Measure-Command { Invoke-WebRequest -Uri $url -UseBasicParsing | Out-Null }

# ~3.7s
Measure-Command { (New-Object System.Net.WebClient).DownloadData($url) }
```

So, I think I'm convinced that just using the simpler `Invoke-WebRequest` is plenty sufficient (and will likely be porting our Server Core files to use it too so that the diff is smaller between variants). :+1: